### PR TITLE
Weather feature

### PR DESCRIPTION
**Pull request by Addis-Belete**
Weather app is a desktop-only app where users can search and get the current weather condition of any city. Is built by fetching data from a weather map.

### CODE REVIEW

**Comment by Addis-Belete** [comment]
Hello TSE
The live link is not working on chrome. so please use another browser to open.
Regards

File: src/dom.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+/* eslint-disable camelcase */
+/* eslint-disable func-names */
+const displayData = function (result) {
```
Comment (SajjadAhmad14) [inline src/dom.js]: Please use the ES6 arrow function instead of the function keyword.

File: src/dom.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  temperature.innerHTML = `${result.main.temp} ℃`;
+  icons.innerHTML = result.weather[0].icon;
+  cityName.innerHTML = result.name;
+  feels.innerHTML = `Feels_like:${result.main.feels_like} ℃ | `;
+  min_temp.innerHTML = `Temp_min:${result.main.temp_min} ℃ | `;
+  max_temp.innerHTML = `Temp_max:${result.main.temp_max} ℃ | `;
+  humidity.innerHTML = `Humidity:${result.main.humidity}`;
+};
+
+const displayError = function () {
```
Comment (SajjadAhmad14) [inline src/dom.js]: Please use the ES6 arrow function instead of the function keyword.

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
+/* eslint-disable camelcase */
+import './style.css';
+import { displayData, displayError } from './dom';
+import clearField from './logic';
+
+const cityWeather = function (city_name = 'Addis Ababa') {
```
Comment (SajjadAhmad14) [inline src/index.js]: Please use the ES6 arrow function instead of the function keyword.

File: src/index.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+      }
+      return response.json();
+    }).then((result) => {
+      displayData(result);
+    }).catch((e) => {
+      console.log(e);
+    });
+};
+
+window.onload = function () {
```
Comment (SajjadAhmad14) [inline src/index.js]: Please use the ES6 arrow function instead of the function keyword.

File: src/logic.js [hunk 5]
```diff
@@ -0,0 +1,5 @@
+/* eslint-disable camelcase */
+export default function clearField() {
```
Comment (SajjadAhmad14) [inline src/logic.js]: Please use the ES6 arrow function instead of the function keyword.

File: src/dom.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
+/* eslint-disable camelcase */
+/* eslint-disable func-names */
```
Comment (SajjadAhmad14) [inline src/dom.js]: Do not disable eslint rules.

File: src/logic.js [hunk 7]
```diff
@@ -0,0 +1,5 @@
+/* eslint-disable camelcase */
```
Comment (SajjadAhmad14) [inline src/logic.js]: Do not disable eslint rules.

File: src/index.js [hunk 8]
```diff
@@ -0,0 +1,40 @@
+/* eslint-disable camelcase */
```
Comment (SajjadAhmad14) [inline src/index.js:1]: Do not disable eslint rules.

File: package.json [hunk 9]
```diff
@@ -0,0 +1,38 @@
+{
+  "name": "weather-app",
+  "version": "1.0.0",
+  "description": "![](https://img.shields.io/badge/Microverse-blueviolet)",
+  "private": true,
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "build": "webpack"
+  },
```
Comment (SajjadAhmad14) [inline package.json]: "scripts": {
    "dev": "webpack --mode development",
    "build": "webpack --mode production",
    "start": "webpack-dev-server --mode development --open" // optional
  },

File: package.json [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+    "stylelint": "^13.13.1",
+    "stylelint-config-standard": "^21.0.0",
+    "stylelint-csstree-validator": "^1.9.0",
+    "stylelint-scss": "^3.19.0",
+    "webpack": "^5.40.0",
+    "webpack-cli": "^4.7.2"
+  },
+  "dependencies": {
+    "lodash": "^4.17.21"
+  }
```
Comment (SajjadAhmad14) [inline package.json]: Please add webpack related packages as `webpack`, `webpack-cli`, and `webpack-dev-server` etc here.

File: package.json [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+    "eslint-config-airbnb-base": "^14.2.1",
+    "eslint-plugin-import": "^2.23.4",
+    "style-loader": "^3.0.0",
+    "stylelint": "^13.13.1",
+    "stylelint-config-standard": "^21.0.0",
+    "stylelint-csstree-validator": "^1.9.0",
+    "stylelint-scss": "^3.19.0",
+    "webpack": "^5.40.0",
+    "webpack-cli": "^4.7.2"
+  },
```
Comment (SajjadAhmad14) [inline package.json]: Please add webpack related package `webpack-dev-server` here.

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
# Changes Required :recycle:
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hello, you have been doing a tremendous job so far :ok_hand:
## To Highlight 🎉
- [ ] You have mastered Git flow.
- [ ] README includes all the instructions to run the project locally.
## To Improve 🙏
- [x] All webpack related packages should be included in package.json like `webpack`, `webpack-cli`, `webpack-dev-server etc`. Also, those packages should be added as devDependencies. Please see this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/devdep_requirement.png).
- [x] Please do not disable eslint rules so much. You should correct errors instead.
- [x] App should have a toggle button to convert Celsius to Fahrenheit and vice versa. Please see this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/search_requirement.png).
- [x] Please use ES6 arrow functions instead of function keyword. 
- [x] Please see inline comments for reference.
# Optional
- [ ] App may fetch the current geolocation of the user when the it is first accessed and fetches the weather from that location
### Happy Coding ☺️
## If you have any questions you can comment here in this PR or can always connect with me on slack ( @Sajjad Ahmad ).

**Comment by Addis-Belete** [comment]
Hello TSE
Thank you for your feedback. I did all the requested changes. kindly review it.
Regards

**Comment by mke2111** [comment]
### Project under review ⚪

**Review by mke2111 — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED 🔴 ♻️ 

## Hey Happy Coder, 👋🏽 
**This is a great project you have going on, well-done lad and thanks for implementing the changes requested earlier on**. 

### Well-done 💯 
    1. The app runs just fine. 💯 
    2. The toggle functionality works as expected. 👍🏽 

**However, there is one section to improve before your project is approved.** 🔴 
<hr>

### REQUIRED

- [x] You are missing the `webpack-dev-server` dependency. Add this to your package JSON file.

#

### Kindly make all the requested changes and submit another CRR. 💯 

**Cheers 🍹 and Happy coding 💻** 

<hr>

Mention @mke2111 in this thread and I'll be glad to help. 😄 
### Best regards
Slack 🆘  Roy Mukuye
Twitter 🐦  [Roy Mukuye](https://twitter.com/Roymkenya)
LinkedIn 🔗  [Roy Mukuye](https://www.linkedin.com/in/roy-mukuye/)

**Comment by Addis-Belete** [comment]
Hello TSE
Thank you for your feedback. I did all the requested changes. kindly review it.
Regards

**Review by Georjane — APPROVED** [review]
## Status: PR Approved ✔️ ✔️
### Hi Team 🙌

Congratulations on successfully completing this project. 🎊 🎊 
Your project looks great and meets all requirements for approval. 

It's a thumbs up for me. 👍 

You can go ahead and merge this PR. 
Good luck on your next project and Happy Coding 💻

**Pull request merged**